src/host/renderData.cpp

```diff
@@ -33,9 +33,12 @@ Microsoft::Console::Types::Viewport RenderData::GetViewport() noexcept
 // - COORD of the end position of the text buffer
 COORD RenderData::GetTextBufferEndPosition() const noexcept
 {
+    // We use the end line of mutableViewport as the end
+    // of the text buffer, it always moves with the written
+    // text
     const CONSOLE_INFORMATION& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
-    Viewport bufferSize = gci.GetActiveOutputBuffer().GetBufferSize();
-    COORD endPosition{ bufferSize.Width() - 1, bufferSize.BottomInclusive() };
+    const auto mutableViewport = gci.GetActiveOutputBuffer().GetVirtualViewport();
+    COORD endPosition{ mutableViewport.Width() - 1, mutableViewport.BottomInclusive() };

```

Comment:
just so i understand: this is _not_ impacted by the user having scrolled up? the viewport _usually_ is.